### ISSUE TITLE
Request to support time-based preview thumbnail sprites

### ISSUE THREAD

**Issue by michaelrlevy** [ableplayer/ableplayer#423]
Some video players support using VTT files to display time-based thumbnail preview images, sometimes referred to as "sprites," so that as users slide their cursor over a time progress bar small rectangles are displayed in the GUI indicating the video content around that time. This is done by creating an image file, or "sprite sheet," consisting of a number of small thumbnail sprites composed into a larger jpg file, associating each of the small thumbnail image rectangles with a time range.

From what I have read, this began with the commercial JW Player and some FOSS players now also provide support.

https://support.jwplayer.com/articles/how-to-add-preview-thumbnails

Video.js seems to support this through at least one plugin:
https://support.brightcove.com/display-thumbnail-previews-plugin

FluidPlayer example:
https://docs.fluidplayer.com/preview_configuration/Radiant Media Player
https://www.radiantmediaplayer.com/docs/latest/preview-thumbnails.html

Here are some code repos for creating the "sprite" thumbnails using ffmpeg

https://github.com/amnuts/jwplayer-thumbnail-preview-generator
https://gist.github.com/lidox/514cd589805944bb31258f856f8cc21f
https://github.com/lidox/mediathek-jwplayer/blob/master/public/js/vtt-creator.js

We make extensive use of sprites in our video display of historical films and oral histories in our collections catalog. I think this would be an extremely useful feature. Of course, we do understand it would require a great deal of effort to implement such a feature. 

https://collections.ushmm.org/search/?f%5Bavailability%5D%5B%5D=digitized&f%5Bf_audiovisual%5D%5B%5D=historic_film

https://collections.ushmm.org/search/?f%5Bavailability%5D%5B%5D=digitized&f%5Bf_audiovisual%5D%5B%5D=video_only&f%5Bf_audiovisual%5D%5B%5D=testimony&page=2

**Comment by terrill** [comment]
@michaelrlevy I agree that this be an extremely useful feature, but I also agree that it would require a great deal of effort. I'm tagging it as a "low priority" enhancement only because I'm prioritizing my own development tasks. Since Able Player's an open source project though, I welcome a pull request from anyone who feels this should be a higher priority.
